errorlint/lint.go

```diff
@@ -880,3 +1018,235 @@ func generateErrorVarName(originalName, typeName string) string {
+func createCombinedDiagnostic(ifStmt *ast.IfStmt, lints []analysis.Diagnostic, extInfo *TypesInfoExt) *analysis.Diagnostic {
+	// Find the earliest position for the combined diagnostic
+	earliestPos := token.NoPos
+	for _, lint := range lints {
+		if earliestPos == token.NoPos || lint.Pos < earliestPos {
+			earliestPos = lint.Pos
+		}
+	}
+
+	// Create the combined diagnostic
+	combined := &analysis.Diagnostic{
+		Pos:     earliestPos,
+		Message: "type assertion and error comparison will fail on wrapped errors. Use errors.As and errors.Is to check for specific errors",
+	}
+
+	// Try to create a combined fix for the if statement
+	suggestedFix := combinedFix(ifStmt, extInfo)
+	if suggestedFix != nil {
+		combined.SuggestedFixes = []analysis.SuggestedFix{*suggestedFix}
+	}
+
+	return combined
+}
+
+// combinedFix creates a suggested fix that handles both type assertion
+// and error comparison in the same if statement.
+// Transforms: if e, ok := err.(*Type); ok && e.Field == value {
+// Into: e := &Type{}; if errors.As(err, &e) && errors.Is(e.Field, value) {
+func combinedFix(ifStmt *ast.IfStmt, extInfo *TypesInfoExt) *analysis.SuggestedFix {
+	// Parse the if statement structure to extract components
+	components := parseIfComponents(ifStmt)
+	if components == nil {
+		return nil
+	}
+
+	// Check if this is an else-if statement
+	components.context.isElseIf = isElseIfStatement(ifStmt, extInfo)
+
+	// Build the replacement text using the extracted components
+	replacement := buildReplacement(components)
+	if replacement == "" {
+		return nil
+	}
+
+	// Determine the replacement range based on whether it's an else-if
+	endPos := ifStmt.Body.Pos()
+	if components.context.isElseIf {
+		// For else-if cases, we need to replace from the "if" to the end of the block
+		// to properly handle the transformation
+		endPos = ifStmt.Body.End()
+	}
+
+	return &analysis.SuggestedFix{
+		Message: "Use errors.As and errors.Is for error handling",
+		TextEdits: []analysis.TextEdit{{
+			Pos:     ifStmt.Pos(),
+			End:     endPos,
+			NewText: []byte(replacement),
+		}},
+	}
+}
+
+// isElseIfStatement checks if the given if statement is part of an else-if construct
+// by checking if it's in the Else field of a parent if statement.
+func isElseIfStatement(ifStmt *ast.IfStmt, extInfo *TypesInfoExt) bool {
+	parent := extInfo.NodeParent[ifStmt]
+	if parent == nil {
+		return false
+	}
+
+	// Check if the parent is an if statement's Else field
+	if parentIf, ok := parent.(*ast.IfStmt); ok {
+		return parentIf.Else == ifStmt
+	}
+
+	return false
+}
+
+// typeAssertion holds type assertion specific data
+type typeAssertion struct {
+	varName    string
+	errorExpr  ast.Expr
+	targetType ast.Expr
+}
+
+// comparison holds error comparison specific data
+type comparison struct {
+	expr    ast.Expr
+	target  ast.Expr
+	negated bool
+}
+
+// context holds if statement context information
+type context struct {
+	isElseIf  bool
+	bodyStmts []ast.Stmt
+}
```

Comment:
It's strange to call it context, especially because it shadows context package

ifInfo maybe?